qutebrowser/components/adblock.py

```diff
@@ -56,80 +63,77 @@ def get_fileobj(byte_io: IO[bytes]) -> IO[bytes]:
         byte_io.seek(0)  # rewind what zipfile.is_zipfile did
         zf = zipfile.ZipFile(byte_io)
         filename = _guess_zip_filename(zf)
-        byte_io = zf.open(filename, mode='r')
+        byte_io = zf.open(filename, mode="r")
     else:
         byte_io.seek(0)  # rewind what zipfile.is_zipfile did
     return byte_io
 
 
-def _is_whitelisted_url(url: QUrl) -> bool:
-    """Check if the given URL is on the adblock whitelist."""
-    for pattern in config.val.content.host_blocking.whitelist:
-        if pattern.matches(url):
-            return True
-    return False
-
-
-class _FakeDownload(downloads.TempDownload):
-
-    """A download stub to use on_download_finished with local files."""
-
-    def __init__(self,  # pylint: disable=super-init-not-called
-                 fileobj: IO[bytes]) -> None:
-        self.fileobj = fileobj
-        self.successful = True
+def _should_be_used() -> bool:
+    """Whether the hostblocker should be used or not."""
+    method = config.val.content.blocking.method
+    adblock_dependency_satisfied = braveadblock.ad_blocker is None
```

Comment:
This seems the wrong way around :wink:

It also means that the `adblock` component relies on `braveadblock` being initialized first, which might not be the case. I'll think about a way to refactor things to make things work - since I'm already in the process of merging this (and also have some other local changes) it's probably easier.